DOC: Windows build instructions

### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)


### Location of the documentation

https://pandas.pydata.org/docs/dev/development/contributing_environment.html#step-1-install-a-c-compiler

### Documentation problem

Installing VS Build Tools 2022 with default settings gives me the following error when I build with mason:

```

(pandas-dev) C:\Users\andre\repos\pandas>python -m pip install -ve . --no-build-isolation --config-settings editable-verbose=true
Using pip 24.2 from C:\Users\andre\mambaforge\envs\pandas-dev\lib\site-packages\pip (python 3.10)
Obtaining file:///C:/Users/andre/repos/pandas
  Running command Checking if build backend supports build_editable
  Checking if build backend supports build_editable ... done
  Running command Preparing editable metadata (pyproject.toml)
  + meson setup C:\Users\andre\repos\pandas C:\Users\andre\repos\pandas\build\cp310 -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --vsenv --native-file=C:\Users\andre\repos\pandas\build\cp310\meson-python-native-file.ini
  The Meson build system
  Version: 1.2.1
  Source dir: C:\Users\andre\repos\pandas
  Build dir: C:\Users\andre\repos\pandas\build\cp310
  Build type: native build
  Project name: pandas
  Project version: 3.0.0.dev0+1427.ge07453e24d

  ..\..\meson.build:2:0: ERROR: Unknown compiler(s): [['cl.exe']]
  The following exception(s) were encountered:
  Running `cl.exe /?` gave "[WinError 2] The system cannot find the file specified"

  A full log can be found at C:\Users\andre\repos\pandas\build\cp310\meson-logs\meson-log.txt
  error: subprocess-exited-with-error

  × Preparing editable metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: 'C:\Users\andre\mambaforge\envs\pandas-dev\python.exe' 'C:\Users\andre\mambaforge\envs\pandas-dev\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py' prepare_metadata_for_build_editable 'C:\Users\andre\AppData\Local\Temp\tmp_6e6694l'
  cwd: C:\Users\andre\repos\pandas
  Preparing editable metadata (pyproject.toml) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

```

### Suggested fix for documentation

Include instructions to select the optional MSVC v142 - VS 2019 C++ x64/x86 build tools on the right of the window:

![image](https://github.com/user-attachments/assets/19dbc141-6397-40b2-ac2b-083dacb76996)

after installing this it compiled.

@andrewgsavage 
Thanks for this information. Thanks to you, I was able to compile pandas on my Windows system. That said, I also encountered an additional issue. 

I couldn't `import pandas` outside the miniforge. When I tried doing so via VS Code and Pycharm, I got the following errors:

```
[11/37] Linking target pandas/_libs/tslibs/conversion.cp310-win_amd64.pyd
FAILED: pandas/_libs/tslibs/conversion.cp310-win_amd64.pyd 
"link"  /MACHINE:x64 /OUT:pandas/_libs/tslibs/conversion.cp310-win_amd64.pyd pandas/_libs/tslibs/conversion.cp310-win_amd64.pyd.p/meson-generated_pandas__libs_tslibs_conversion.pyx.c.obj "/nologo" "/release" "/nologo" "/OPT:REF" "/DLL" "/IMPLIB:pandas\_libs\tslibs\conversion.cp310-win_amd64.lib" "C:\Users\Ammar\.conda\envs\pandas-dev\libs\python310.lib" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
LINK : fatal error LNK1104: cannot open file 'pandas\_libs\tslibs\conversion.cp310-win_amd64.pyd'
[22/37] Compiling C object pandas/_libs/index.cp310-win_amd64.pyd.p/meson-generated_pandas__libs_index.pyx.c.obj
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "d:\Programming\Open Source Projects\Pandas\pandas-ammar\tempCodeRunnerFile.python", line 1, in <module>
    import pandas
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1002, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 945, in _find_spec
  File "C:\Users\Ammar\.conda\envs\pandas-dev\lib\site-packages\_pandas_editable_loader.py", line 268, in find_spec
    tree = self.rebuild()
  File "C:\Users\Ammar\.conda\envs\pandas-dev\lib\site-packages\_pandas_editable_loader.py", line 309, in rebuild
    subprocess.run(self._build_cmd, cwd=self._build_path, env=env, stdout=stdout, check=True)
  File "C:\Users\Ammar\.conda\envs\pandas-dev\lib\subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['meson', 'compile']' returned non-zero exit status 1.
``` 
While I couldn't find any documentation on the error, there was a comment on Stackflow, suggesting users to do `pip install cython` for resolving pandas import issues. I did the same, and my problem got resolved. 

What's the reason for that? Isn't cython already present in the virtual environment?

--- a/doc/source/development/contributing_environment.rst
+++ b/doc/source/development/contributing_environment.rst
@@ -35,6 +35,10 @@ You will need `Build Tools for Visual Studio 2022
         scrolling down to "All downloads" -> "Tools for Visual Studio".
         In the installer, select the "Desktop development with C++" Workloads.
 
+        If you encounter an error indicating `cl.exe` is not found when building with Meson, 
+        reopen the installer and also select the optional component 
+        **MSVC v142 - VS 2019 C++ x64/x86 build tools** in the right pane for installation.
+
 Alternatively, you can install the necessary components on the commandline using
 `vs_BuildTools.exe <https://learn.microsoft.com/en-us/visualstudio/install/use-command-line-parameters-to-install-visual-studio?source=recommendations&view=vs-2022>`_
 

--- a/doc/source/development/contributing_environment.rst
+++ b/doc/source/development/contributing_environment.rst
@@ -35,8 +35,8 @@ You will need `Build Tools for Visual Studio 2022
         scrolling down to "All downloads" -> "Tools for Visual Studio".
         In the installer, select the "Desktop development with C++" Workloads.
 
-        If you encounter an error indicating `cl.exe` is not found when building with Meson, 
-        reopen the installer and also select the optional component 
+        If you encounter an error indicating ``cl.exe`` is not found when building with Meson,
+        reopen the installer and also select the optional component
         **MSVC v142 - VS 2019 C++ x64/x86 build tools** in the right pane for installation.
 
 Alternatively, you can install the necessary components on the commandline using

